coffeecard/CoffeeCard.Library/Services/v2/IPurchaseService.cs

```diff
@@ -38,5 +38,12 @@ public interface IPurchaseService : IDisposable
         /// <param name="webhook">Webhook data object</param>
         /// <returns></returns>
         Task HandleMobilePayPaymentUpdate(MobilePayWebhook webhook);
+
+        /// <summary>
+        /// Redeem af voucher code for a purchase
+        /// </summary>
+        /// <param name="voucherCode">Voucher code</param>
+        /// <param name="user">user redeeming the voucher</param>
+        Task<SimplePurchaseResponse> RedeemVoucher(string voucherCode, User user);
```

Comment:
I have been thinking is this would not make more sense to have in IVoucherService, what were you thoughts?

Reply:
Hi @fredpetersen , 

@marfavi and I discussed this. Omid's argumentation is that not all Purchases have a Payment and hence strictly speaking not a purchase. This is for instance the case with vouchers where a user is not _purchasing_ anything but he's using a voucher which grants him Tickets. The point is that our data model is not reflecting this and it would have been best if it did. However this will require both a larger redesign and secondly a data migration. So create an issue from this and you can go ahead and merge this Pull Request. Yay 🙌🏻